python/plugins/processing/tools/vector.py

```diff
@@ -521,6 +523,21 @@ def ogrLayerName(uri):
         regex = re.compile('(layername=)(.*)')
         r = regex.search(uri)
         return r.groups()[1]
+    elif '.gdb' in uri:
```

Comment:
It is to handle the case in which there's no "|layerid" nor "|layername".
Can that case be generalized ?